### PR TITLE
set context class loader to AgentClassLoader

### PR DESCRIPTION
**Pull request by DuanYuePeng**
javax.security.auth.login.LoginContext use context class loader to load the login module, when configuring "sasl.jaas.config", we could set the value as `org.apache.skywalking.apm.dependencies.org.apache.kafka.common.security.scram.ScramLoginModule required username="user" password="pass";`, and the context class loader would load the class correctly, otherwise, NoClassFoundException would be thrown

### Fix <https://github.com/apache/skywalking/issues/6157>
- [ ] Add a unit test to verify that the fix works.
- [x] Explain briefly why the bug exists and how to fix it.

- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [ ] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/loader/AgentClassLoader.java [hunk 1]
```diff
@@ -77,6 +77,7 @@ public static void initDefaultLoader() throws AgentPackageNotFoundException {
             synchronized (AgentClassLoader.class) {
                 if (DEFAULT_LOADER == null) {
                     DEFAULT_LOADER = new AgentClassLoader(PluginBootstrap.class.getClassLoader());
+                    Thread.currentThread().setContextClassLoader(DEFAULT_LOADER);
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/plugin/loader/AgentClassLoader.java:80]: This is only work for current thread, why this effects the Kafka kernel? How they share the same thread?

**Comment by DuanYuePeng** [comment]
This bug happens in `KafkaProducerManager::prepare` method, which is a `BootService`. It works in the same thread with `SkyWalkingAgent::premain`.
```
AdminClient adminClient = AdminClient.create(properties);
```
Just see the Thread dump.
```
init:278, LoginContext (javax.security.auth.login)
<init>:512, LoginContext (javax.security.auth.login)
login:59, AbstractLogin (org.apache.skywalking.apm.dependencies.org.apache.kafka.common.security.authenticator)
<init>:62, LoginManager (org.apache.skywalking.apm.dependencies.org.apache.kafka.common.security.authenticator)
acquireLoginManager:105, LoginManager (org.apache.skywalking.apm.dependencies.org.apache.kafka.common.security.authenticator)
configure:147, SaslChannelBuilder (org.apache.skywalking.apm.dependencies.org.apache.kafka.common.network)
create:146, ChannelBuilders (org.apache.skywalking.apm.dependencies.org.apache.kafka.common.network)
clientChannelBuilder:67, ChannelBuilders (org.apache.skywalking.apm.dependencies.org.apache.kafka.common.network)
createChannelBuilder:99, ClientUtils (org.apache.skywalking.apm.dependencies.org.apache.kafka.clients)
createInternal:426, KafkaAdminClient (org.apache.skywalking.apm.dependencies.org.apache.kafka.clients.admin)
create:59, Admin (org.apache.skywalking.apm.dependencies.org.apache.kafka.clients.admin)
create:39, AdminClient (org.apache.skywalking.apm.dependencies.org.apache.kafka.clients.admin)
prepare:58, KafkaProducerManager (org.apache.skywalking.apm.agent.core.kafka)
lambda$prepare$1:105, ServiceManager (org.apache.skywalking.apm.agent.core.boot)
accept:-1, 783191662 (org.apache.skywalking.apm.agent.core.boot.ServiceManager$$Lambda$6)
accept:183, ForEachOps$ForEachOp$OfRef (java.util.stream)
end:357, SortedOps$SizedRefSortingSink (java.util.stream)
copyInto:483, AbstractPipeline (java.util.stream)
wrapAndCopyInto:472, AbstractPipeline (java.util.stream)
evaluateSequential:150, ForEachOps$ForEachOp (java.util.stream)
evaluateSequential:173, ForEachOps$ForEachOp$OfRef (java.util.stream)
evaluate:234, AbstractPipeline (java.util.stream)
forEach:485, ReferencePipeline (java.util.stream)
prepare:103, ServiceManager (org.apache.skywalking.apm.agent.core.boot)
boot:44, ServiceManager (org.apache.skywalking.apm.agent.core.boot)
premain:133, SkyWalkingAgent (org.apache.skywalking.apm.agent)
invoke0:-1, NativeMethodAccessorImpl (sun.reflect)
invoke:62, NativeMethodAccessorImpl (sun.reflect)
invoke:43, DelegatingMethodAccessorImpl (sun.reflect)
invoke:498, Method (java.lang.reflect)
loadClassAndStartAgent:386, InstrumentationImpl (sun.instrument)
loadClassAndCallPremain:401, InstrumentationImpl (sun.instrument)
```

**Comment by wu-sheng** [comment]
> This bug happens in KafkaProducerManager::prepare method, which is a BootService. It works in the same thread with SkyWalkingAgent::premain.

Then, I read the codes, then this is not premain thread, this is a thread created through `Executors.newSingleThreadScheduledExecutor`. You should not manipulate the premain thread class loader, instead, please use this inside `KafkaProducerManager#run` when it runs in the first time, which is much safer and only effects a thread we created inside the agent.

Does this make sense for you? Could you try locally?

**Comment by wu-sheng** [comment]
And you should update the `changes.md` in the root.

**Comment by wu-sheng** [comment]
You seems to change a lot of unexpected things. Please recheck your local git.

**Comment by DuanYuePeng** [comment]
The git is broken, I'll start a new PR.

**Pull request closed without merging**